landing page on react

roman454647 opened the pull request.

RomanPidkostelnyi committed:
landing page on react

anna-kuzii commented on package.json:
```diff
@@ -0,0 +1,43 @@
+    "start": "react-scripts start",
+    "build": "react-scripts build",
+    "test": "react-scripts test",
+    "eject": "react-scripts eject"
```
I believe, we agreed to have precommit with linters. Pls add it

anna-kuzii commented on src/About/index.jsx:
```diff
@@ -0,0 +1,42 @@
+import './style.scss';
+import divider from '../image/Divider.png';
+import cookImg from '../image/cook.png';
+import dish from '../image/dish.png';
```
Usually, fonts, imgs, etc are added to assets folder

roman454647 replied:
done

anna-kuzii commented on src/About/index.jsx:
```diff
@@ -0,0 +1,42 @@
+import dish from '../image/dish.png';
+
+function About() {
+  return (
```
pls use es6

roman454647 replied:
i have used pure function like  About =() =>

anna-kuzii commented on src/About/index.jsx:
```diff
@@ -0,0 +1,42 @@
+
+function About() {
+  return (
+    <section id="about" className="wrapper">
```
class wrapper is to general. do u use it on the other components?

roman454647 replied:
i have changed the name of wrapper

anna-kuzii commented on src/About/style.scss:
```diff
@@ -0,0 +1,96 @@
+  background-position: center center;
+  background-repeat: no-repeat;
+  background-size: cover;
+}
```
why so long? can u combine them?

roman454647 replied:
yes , i have combined  it in one line

anna-kuzii commented on src/About/style.scss:
```diff
@@ -0,0 +1,96 @@
+@import './_variables.scss';
+
+.wrapper {
```
if class wrapper is used in a few places and its styles are the same it should be replaced in general.scss. Otherwise pls change the name to more appropriate

roman454647 replied:
i have changed the name of wrapper

anna-kuzii commented on src/About/style.scss:
```diff
@@ -0,0 +1,96 @@
+
+  h1 {
+    padding-bottom: 13px;
+    font-size: 36px;
```
it is better to use rem for font size. Then for mobile it will look better

roman454647 replied:
ok, it  was changed

anna-kuzii commented on src/About/style.scss:
```diff
@@ -0,0 +1,96 @@
+}
+
+
+.mw-100 {
```
move it to general.scss

roman454647 replied:
i have changed that part of code

anna-kuzii commented on src/About/style.scss:
```diff
@@ -0,0 +1,96 @@
+  min-width: 100%;
+}
+
+img {
```
does it influence to all imgs?

roman454647 replied:
i have changed that block of code to another class

anna-kuzii commented on src/About/style.scss:
```diff
@@ -0,0 +1,96 @@
+  padding-top: 18px;
+  margin-left: 33%;
+}
+@media screen and (max-width: 1414px) {
```
pls use var
also pls use more general breakpoints

roman454647 replied:
i changed  all resolution to variable

anna-kuzii commented on src/About/style.scss:
```diff
@@ -0,0 +1,96 @@
+
+.cook {
+  padding-top: 18px;
+  margin-left: 33%;
```
hmmm...I am not sure about this margin. Do u make it on the center using it?

roman454647 replied:
not, so i changed it

anna-kuzii commented on src/About/style.scss:
```diff
@@ -0,0 +1,96 @@
+  }
+}
+@media screen and (max-width: 1278px) {
+  .mw-100 {
```
mw means min-width, not min-width+margin-left. Change it pls



Also it is better to use media query INSIDE selector, but not selector inside media query.

Pls change it

roman454647 replied:
changed

anna-kuzii commented on src/About/style.scss:
```diff
@@ -0,0 +1,96 @@
+
+  .about-right {
+    margin-left: 0;
+    text-align: center;
```
u can combine this class and previous one

roman454647 replied:
combined)

anna-kuzii commented on src/About/style.scss:
```diff
@@ -0,0 +1,96 @@
+      margin-left: 0;
+    }
+  }
+  @media screen and (max-width: 900px) {
```
media query inside media query???

roman454647 replied:
my fault ,changed

anna-kuzii commented on src/App.jsx:
```diff
@@ -0,0 +1,24 @@
+import Rewiews from './Rewiews';
+
+function App() {
+  return (
```
es6 pls

roman454647 replied:
ok)

anna-kuzii commented on src/BookTable/index.jsx:
```diff
@@ -0,0 +1,67 @@
+      <div className="book_form">
+        <div className="left-side">
+          <img src={image3} alt="im3" />
+          <img src={image4} alt="im4" />
```
alt should include string which has understandable text for person and screen readers. Pls change it

roman454647 replied:
i have changed it to more understandable text

anna-kuzii commented on src/BookTable/index.jsx:
```diff
@@ -0,0 +1,67 @@
+          <form name="bookform" action="#" method="post">
+            <div className="form_inputs">
+              <div className="right_inputs">
+                <p>Name</p>
```
pls use label with input , not p

roman454647 replied:
i changed all p to label

anna-kuzii commented on src/BookTable/style.scss:
```diff
@@ -0,0 +1,136 @@
+  flex-direction: row;
+  justify-content: flex-start;
+  max-height: 80%;
+  padding: 120px 0 100px;
```
are u sure in it?

roman454647 replied:
yes)

anna-kuzii commented on src/Footer/style.scss:
```diff
@@ -0,0 +1,74 @@
+  }
+}
+
+.About {
```
why this class name is different the other one?

roman454647 replied:
i have changed the name

anna-kuzii commented on src/Footer/style.scss:
```diff
@@ -0,0 +1,74 @@
+}
+
+.Our-Location {
+  margin-left: 190px;
```
too big margin! are u sure in both of them??

roman454647 replied:
not really ,so i  have 
changed it to 100px

anna-kuzii replied:
still to much
I will double check it on my local

anna-kuzii commented on src/Header/style.scss:
```diff
@@ -0,0 +1,7 @@
+.container {
```
is this general class?

roman454647 replied:
no it just for promo and navigation bar ,so i changed name to more understandable

anna-kuzii commented on src/Ingredients/index.jsx:
```diff
@@ -0,0 +1,39 @@
+import divider1 from '../image/Divider1.png';
+import im from '../image/image.png';
+import im1 from '../image/image1.png';
+import im2 from '../image/image2.png';
```
really bad names!

roman454647 replied:
changed

anna-kuzii commented on src/Ingredients/style.scss:
```diff
@@ -0,0 +1,95 @@
+
+.section-ingredients {
+  padding-top: 170px;
+  padding-bottom: 150px;
```
too big. are u sure?

roman454647 replied:
was changed to smaller

anna-kuzii commented on src/Ingredients/style.scss:
```diff
@@ -0,0 +1,95 @@
+    font-size: 36px;
+    padding-bottom: 17px;
+    color: $white;
+    margin-left: 59.5%;
```
too big + bad value. Why not 60%?

roman454647 replied:
changed to 60%

anna-kuzii commented on src/Ingredients/style.scss:
```diff
@@ -0,0 +1,95 @@
+
+
+
+.divider2 {
```
bad class name

roman454647 replied:
changed to more general

anna-kuzii commented on src/Ingredients/style.scss:
```diff
@@ -0,0 +1,95 @@
+.divider2 {
+  justify-content: center;
+  max-width: 100%;
+  margin-left: 60%;
```
it is bigger than half. What do u do with that? what effect doest it give u?

anna-kuzii commented on src/Menu/index.jsx:
```diff
@@ -0,0 +1,145 @@
+
+
+function Menu() {
+  return (
```
too long component with the same divs. Can we move data to Array and use map function for example and render it?

roman454647 replied:
changed to function in another file

anna-kuzii commented on src/Menu/style.scss:
```diff
@@ -0,0 +1,109 @@
+    padding-top: 15px;
+  }
+
+  .tittle {
```
title

roman454647 replied:
changed to title

anna-kuzii commented on src/Menu/style.scss:
```diff
@@ -0,0 +1,109 @@
+  .special {
+    margin-top: 15px;
+    height: 70%;
+    border: 2px solid rgba(204, 153, 0, 0.3);
```
use var here

roman454647 replied:
used

anna-kuzii commented on src/Menu/style.scss:
```diff
@@ -0,0 +1,109 @@
+    margin-left: 10px;
+    padding-top: 17px;
+    font-size: 21px;
+    color: #cc9900;
```
and here

roman454647 replied:
done

anna-kuzii commented on src/Menu/style.scss:
```diff
@@ -0,0 +1,109 @@
+  .price {
+    text-align: right;
+    margin-right: auto;
+    float: right;
```
can we use flex instead of float?

roman454647 replied:
changed to flex

anna-kuzii commented on src/Menu/style.scss:
```diff
@@ -0,0 +1,109 @@
+    padding-top: 10px;
+    font-size: 16px;
+    padding-left: 10px;
+    padding-bottom: 10px;
```
can we combine all paddings?

roman454647 replied:
combined

anna-kuzii commented on src/Menu/style.scss:
```diff
@@ -0,0 +1,109 @@
+
+
+.right-part {
+  margin-left: 170px;
```
too big

roman454647 replied:
changed

anna-kuzii commented on src/Navbar/index.jsx:
```diff
@@ -0,0 +1,35 @@
+          <a href="#menu">Menu</a>
+          <a href="#rewiews">Reviews</a>
+          <a href="#book">Reservations</a>
+          <a href="#rewiews">
```
we can use array here

roman454647 replied:
used

anna-kuzii commented on src/Navbar/style.scss:
```diff
@@ -0,0 +1,91 @@
+  a {
+    color: $white;
+    text-align: center;
+    padding: 22px 24px;
```
I believe u should have a lot of warnings in the lint. Do u?

roman454647 replied:
now all clear

anna-kuzii commented on src/Promo/index.jsx:
```diff
@@ -0,0 +1,36 @@
+import React from 'react';
+import './style.scss';
+import btnBlack from '../image/button-black.png';
+import btnWhite from '../image/Button White.png';
```
buttons using img? Pls do it using your hands

roman454647 replied:
done

anna-kuzii commented on src/Promo/index.jsx:
```diff
@@ -0,0 +1,36 @@
+      <div className="buttons">
+        <input
+          type="image"
+          className="white"
```
bad name

roman454647 replied:
changed

anna-kuzii commented on src/Promo/style.scss:
```diff
@@ -0,0 +1,48 @@
+  flex-direction: column;
+  flex-wrap: wrap;
+  width: 100%;
+  padding: 10px 0 100px;
```
bad value

roman454647 replied:
why?

anna-kuzii replied:
too big

anna-kuzii commented on src/Rewiews/index.jsx:
```diff
@@ -0,0 +1,30 @@
+function Rewiews() {
+  return (
+    <section id="rewiews" className="wrapper-rewiews">
+      <div className="Rewiews" style={{ 'max-width': '900px' }}>
```
why inline?

roman454647 replied:
changed

anna-kuzii commented on src/Rewiews/index.jsx:
```diff
@@ -0,0 +1,30 @@
+          <h1>Guest Reviews</h1>
+          <img src={divSlider} alt="slider" />
+          <p>
+            f you’ve been to one of our restaurants, you’ve seen – and
```
f?

roman454647 replied:
if

anna-kuzii commented on src/Rewiews/style.scss:
```diff
@@ -0,0 +1,82 @@
+  padding-top: 78px;
+  padding-bottom: 100px;
+
+  color: $white;
```
u missed some code because of empty lines?

roman454647 replied:
changed

anna-kuzii commented on src/variables/_code.scss:
```diff
@@ -0,0 +1,3 @@
+$white: #ffffff;
```
why this vars in a separate file not in variables.scss?

roman454647 replied:
i have removed that file

anna-kuzii commented on src/variables/_code.scss:
```diff
@@ -0,0 +1,3 @@
+$white: #ffffff;
+$ash: #c6c3b9;
+$black: #000000;
```
u can delete yarn, test files, right?

roman454647 replied:
deleted already

RomanPidkostelnyi committed:
update

eplied to almost all comments  ,except :
- photo in different folder
- menu component is too big
- and package.json

roman454647 committed:
Update README.md

roman454647 commented:
TODO
-import all photos in one file
-rewrite menu to array
-rewrite buttons 
-write precommit in package.json

anna-kuzii commented on src/About/style.scss:
```diff
@@ -0,0 +1,86 @@
+    margin-left: 0;
+  }
+
+  @media screen and (max-width: 900px) {
```
var?

roman454647 replied:
used

anna-kuzii commented on src/About/style.scss:
```diff
@@ -0,0 +1,86 @@
+  margin-left: 20%;
+}
+@media screen and (max-width: $wide-medium-screen) {
+  .About {
```
media inside selector

roman454647 replied:
changed

anna-kuzii commented on src/BookTable/index.jsx:
```diff
@@ -0,0 +1,77 @@
+  <section id="book" className="book_wrapper">
+    <div className="book_form">
+      <div className="left-side">
+        <img src={image3} alt="nuts with chery" />
```
cherry :)

roman454647 replied:
cherry,now

anna-kuzii commented on src/BookTable/index.jsx:
```diff
@@ -0,0 +1,77 @@
+        </p>
+
+        <p>
+          {''}
```
what is that?

roman454647 replied:
even i don`t know )

anna-kuzii commented on src/BookTable/index.jsx:
```diff
@@ -0,0 +1,77 @@
+              type="image"
+              src={bookNow}
+              value="Book now!"
+              alt="bok_now_button"
```
pls make alt suitable

roman454647 replied:
i think i have made it

anna-kuzii commented on src/BookTable/style.scss:
```diff
@@ -0,0 +1,124 @@
+  justify-content: center;
+  padding-bottom: 30px;
+
+  @media screen and (max-width: 1278px) {
```
var

roman454647 replied:
changed

anna-kuzii commented on src/BookTable/style.scss:
```diff
@@ -0,0 +1,124 @@
+}
+
+.left-side {
+  @media screen and (max-width: 1414px) {
```
var

roman454647 replied:
changed

roman454647 committed:
adding registration

adding authentification  using
React ,Redux, Node Js ,Express

roman454647 committed:
Update README.md

anna-kuzii commented on .eslintrc:
```diff
@@ -0,0 +1,7 @@
+  "extends": "eslint-config-airbnb",
+  "env": {
+    "browser": true
+    },
```
any rules?

anna-kuzii commented on .sass-lint.yml:
```diff
@@ -0,0 +1,97 @@
+  no-attribute-selectors: 0
+  no-color-hex: 0
+  no-color-keywords: 1
+  no-color-literals: 1
```
we agreed to change all of them to error

roman454647 replied:
changed to error

anna-kuzii commented on README.md:
```diff
@@ -1,2 +1,8 @@
+site for  Lambda restorant on React
+to run application you need to
+-clone git repository
+-from react-site folder run nmp install
```
npm not nmp

roman454647 replied:
npm

anna-kuzii commented on backend/app.js:
```diff
@@ -0,0 +1,31 @@
+app.use('/api/users', users);
+
+app.get('/', function(req, res) {
+    res.send('hello');
```
Hello

roman454647 replied:
Hello

InnaKutsik commented on backend/db.js:
```diff
@@ -0,0 +1,3 @@
+module.exports = {
+  DB: 'mongodb://localhost:27017/auth',
```
move to env file

roman454647 replied:
done

anna-kuzii commented on backend/routes/user.js:
```diff
@@ -0,0 +1,124 @@
+              }
+            });
+          } else {
+            console.log("incorrext password")
```
incorrext?

roman454647 replied:
incorrect

InnaKutsik commented on backend/routes/user.js:
```diff
@@ -0,0 +1,124 @@
+  } = validateRegisterInput(req.body);
+
+  if (!isValid) {
+    console.log("error in 20 lines of code user.js")
```
?

roman454647 replied:
if you about that console log ,I have already deleted it

anna-kuzii commented on src/About/style.scss:
```diff
@@ -0,0 +1,86 @@
+.section-about {
+  display: flex;
+  width: 70%;
+  padding: 20px 0 20px;
```
I believe linter disagrees with u about that

roman454647 replied:
yes,changed

anna-kuzii commented on src/About/style.scss:
```diff
@@ -0,0 +1,86 @@
+  display: flex;
+  width: 70%;
+  padding: 20px 0 20px;
+  margin: 0 auto;
```
u can make it on center using flex

roman454647 replied:
done

anna-kuzii commented on src/AboutUser/index.jsx:
```diff
@@ -0,0 +1,40 @@
+      <div>
+          <h1>HI, Here some information about you</h1>
+          <h1>It is your name</h1>
+           {user.name}
```
should it be in a selector? p or h4 or span?

roman454647 replied:
it was just for example ,that server side working

anna-kuzii commented on src/AllMenu/index.jsx:
```diff
@@ -0,0 +1,93 @@
+
+  toggleChangeMJ(){
+     this.setState(prevState => ({
+       isMJ: !prevState.isMJ,
```
what is MJ and JB?

roman454647 replied:
it was example from some tutoril how to save text from checkbox, so i deleted it

anna-kuzii commented on src/AllMenu/index.jsx:
```diff
@@ -0,0 +1,93 @@
+    e.preventDefault();
+   let arr = [];
+   for (var key in this.state) {
+     if(this.state[key] === true) {
```
pls rewrite it

roman454647 replied:
deleted folder

anna-kuzii commented on src/App.jsx:
```diff
@@ -0,0 +1,37 @@
+    store.dispatch(logoutUser());
+    window.location.href = '/login';
+  }
+}
```
move it to a genral file

roman454647 replied:
like function?

InnaKutsik commented on backend/routes/user.js:
```diff
@@ -0,0 +1,124 @@
+
+const User = require('../models/User');
+
+router.post('/register', function(req, res) {
```
too long callback function

For example, create some controls folder, create register controller and move there callback ib separate function.
+ split function. For example,
- call to backend (if email exist)
- register bussines logic.

roman454647 replied:
done

anna-kuzii commented on src/App.jsx:
```diff
@@ -0,0 +1,37 @@
+const App = () => (
+  <Provider store={store}>
+    <Router>
+      <div>
```
why do u need div here?

roman454647 replied:
removed

anna-kuzii commented on src/BookTable/index.jsx:
```diff
@@ -0,0 +1,77 @@
+      </div>
+      <div className="right-side">
+        <h1>Just the right food</h1>
+        <img src={divider4} alt="div4" />
```
alt

roman454647 replied:
changed to more understable

InnaKutsik commented on backend/routes/user.js:
```diff
@@ -0,0 +1,124 @@
+  });
+});
+
+router.post('/login', (req, res) => {
```
too long callback function

For example, create some controls folder, create login controller and move there callback in separate function.

split function. For example,
call to backend (if email exist)
register bussines logic.

roman454647 replied:
done

anna-kuzii commented on src/BookTable/index.jsx:
```diff
@@ -0,0 +1,77 @@
+  </section>
+);
+
+export default BookTable;
```
pls export it near const

roman454647 replied:
export near the function

anna-kuzii commented on src/BookTable/style.scss:
```diff
@@ -0,0 +1,124 @@
+  }
+}
+
+.left-side {
```
pls use one convention _ or -

roman454647 replied:
changed all to -

anna-kuzii commented on src/Footer/index.jsx:
```diff
@@ -0,0 +1,50 @@
+          <br />
+          128 Meserole Avenue
+        </p>
+        <img src={icons} alt="if" />
```
alt

roman454647 replied:
changed

InnaKutsik commented on backend/routes/user.js:
```diff
@@ -0,0 +1,124 @@
+  const email = req.body.email;
+  const password = req.body.password;
+
+  User.findOne({
```
you already used this method in previous callback

pls move to separate file and method if it is reusable.
For example, dbQuery folder, user file, and method findByEmail or something like this

anna-kuzii commented on src/Header/style.scss:
```diff
@@ -0,0 +1,3 @@
+.container-home {
+  background: fixed url('../image/Bg2.png') no-repeat center center / cover;
```
what does fixed do?

roman454647 replied:
it makes a cool effect that page and background doesn`t connected ,when you scroll the page

anna-kuzii replied:
great. thx

InnaKutsik commented on backend/routes/user.js:
```diff
@@ -0,0 +1,124 @@
+              avatar: user.avatar
+            }
+            jwt.sign(payload, 'secret', {
+              expiresIn: 3600
```
magic number 
move to constant

roman454647 replied:
moved

anna-kuzii commented on src/Ingredients/style.scss:
```diff
@@ -0,0 +1,83 @@
+  display: flex;
+  flex-direction: column;
+  padding-top: 100px;
+  padding-bottom: 100px;
```
are u sure in these values?

roman454647 replied:
yes

anna-kuzii commented on src/Ingredients/style.scss:
```diff
@@ -0,0 +1,83 @@
+  padding-bottom: 100px;
+  background-image: url("../image/bg4.png");
+  background-position: 73%;
+  background-repeat: no-repeat;
```
combine it pls

roman454647 replied:
combined

anna-kuzii commented on src/Ingredients/style.scss:
```diff
@@ -0,0 +1,83 @@
+
+  p {
+    line-height: 1.5;
+    font-size: 16px;
```
we agreed to use rem

roman454647 replied:
changed

anna-kuzii commented on src/Login/index.jsx:
```diff
@@ -0,0 +1,114 @@
+    render() {
+        const {errors} = this.state;
+        return(
+        <div className="container" style={{ marginTop: '50px', width: '700px'}}>
```
why inline?

roman454647 replied:
becouse it is just for server side,i will change

anna-kuzii commented on src/Login/index.jsx:
```diff
@@ -0,0 +1,114 @@
+                    <input
+                    type="email"
+                    placeholder="Email"
+                    className={classnames('form-control form-control-lg', {
```
if u use classname then use it everywhere. Thx . Like it

roman454647 replied:
ok

anna-kuzii commented on src/Menu/index.jsx:
```diff
@@ -0,0 +1,126 @@
+            Pastitsio
+            <span className="price">$5.99</span>
+          </h1>
+          <img src={imageLine} alt="line" />
```
array pls

roman454647 replied:
changed

InnaKutsik commented on src/AllMenu/index.jsx:
```diff
@@ -0,0 +1,93 @@
+    }
+  }
+
+  toggleChangeMJ(){
```
MJ ?

roman454647 replied:
folder was deleted

InnaKutsik commented on src/AllMenu/index.jsx:
```diff
@@ -0,0 +1,93 @@
+     }));
+   }
+
+   toggleChangeJB() {
```
JB ?

roman454647 replied:
i have deleted that file

anna-kuzii commented on src/Navbar/index.jsx:
```diff
@@ -0,0 +1,66 @@
+      )
+
+        return(
+    <header>
```
hard to read this code. pls use ctrl+alt+L

InnaKutsik commented on src/AllMenu/index.jsx:
```diff
@@ -0,0 +1,93 @@
+     if(this.state[key] === true) {
+       arr.push(key);
+     }
+   }
```
pls use filter method instead of for/in cycle

roman454647 replied:
folder was deleted

InnaKutsik commented on src/AllMenu/index.jsx:
```diff
@@ -0,0 +1,93 @@
+     }
+   }
+   let data = {
+     check: arr.toString()
```
arr.toString() - reason?

roman454647 replied:
file  deleted ,already

anna-kuzii commented on src/actions/types.js:
```diff
@@ -0,0 +1,2 @@
+export const GET_ERRORS = 'GET_ERRORS';
```
actions should be inside components folder

roman454647 replied:
what does it mean? that every component need  to have his own action?

anna-kuzii replied:
yes :)

anna-kuzii commented on src/navbarList.js:
```diff
@@ -0,0 +1,32 @@
+export default [
+  {
```
it can be in JSON format and inside component's folder

roman454647 replied:
changed location

InnaKutsik commented on src/App.jsx:
```diff
@@ -0,0 +1,37 @@
+  const currentTime = Date.now() / 1000;
+  if (decoded.exp < currentTime) {
+    store.dispatch(logoutUser());
+    window.location.href = '/login';
```
could use some route redux methods?

roman454647 replied:
yes, react -router-redux has a push method

anna-kuzii commented on src/variables/_variables.scss:
```diff
@@ -0,0 +1,10 @@
+$hokey-pokey: #c9a131;
+$black: #000000;
+$l-screen: 1660px;
+$budha-border: rgba(204, 153, 0, 0.3);
```
usually we use

$max-xs-screen: ...
$max-sm-screen: ...
$max-md-screen: ...

$min-sm-screen: $max-xs-screen + 1;
$min-md-screen: $max-sm-screen + 1;
$min-lg-screen: $max-md-screen + 1;

roman454647 replied:
changed

InnaKutsik commented on src/Login/index.jsx:
```diff
@@ -0,0 +1,114 @@
+    render() {
+        const {errors} = this.state;
+        return(
+        <div className="container" style={{ marginTop: '50px', width: '700px'}}>
```
inline styles?

roman454647 replied:
changed to style.scss

InnaKutsik commented on src/Login/index.jsx:
```diff
@@ -0,0 +1,114 @@
+        const {errors} = this.state;
+        return(
+        <div className="container" style={{ marginTop: '50px', width: '700px'}}>
+            <h2 style={{marginBottom: '40px'}}>Login</h2>
```
inline style?

roman454647 replied:
style.scss was created)

InnaKutsik commented on src/Login/index.jsx:
```diff
@@ -0,0 +1,114 @@
+                    <input
+                    type="email"
+                    placeholder="Email"
+                    className={classnames('form-control form-control-lg', {
```
move expression to variable and use here

roman454647 replied:
moved to variable

InnaKutsik commented on src/Login/index.jsx:
```diff
@@ -0,0 +1,114 @@
+                    <input
+                    type="password"
+                    placeholder="Password"
+                    className={classnames('form-control form-control-lg', {
```
move expression to variable and use here

roman454647 replied:
moved to variable

InnaKutsik commented on src/Navbar/index.jsx:
```diff
@@ -0,0 +1,66 @@
+            <Fragment>
+              <a href=""  onClick={this.onLogout.bind(this)}>
+                    <img src={user.avatar} alt={user.name} title={user.name}
+                        style={{ width: '25px', marginRight: '5px'}} />
```
inline styles?

roman454647 replied:
changed

InnaKutsik commented on src/Register/index.jsx:
```diff
@@ -0,0 +1,141 @@
+
+    handleSubmit(e) {
+        e.preventDefault();
+        console.log('name:'+this.state.email);
```
?

roman454647 replied:
it was for debug ,i have deleted already

InnaKutsik commented on src/Register/index.jsx:
```diff
@@ -0,0 +1,141 @@
+    render() {
+        const { errors } = this.state;
+        return(
+        <div className="container" style={{ marginTop: '50px', width: '700px'}}>
```
inline styles?

roman454647 replied:
changed

InnaKutsik commented on src/Rewiews/style.scss:
```diff
@@ -0,0 +1,70 @@
+  background:  url('../image/bg6_slider.png') no-repeat center ;
+  text-align: center;
+  margin-left: 30%;
+  padding-top: 130px;
```
so huge number?

roman454647 replied:
yes ,now it 100

anna-kuzii replied:
no so small))

InnaKutsik commented on src/actions/authentication.js:
```diff
@@ -0,0 +1,48 @@
+import setAuthToken from '../setAuthToken';
+
+
+export const registerUser = (user, history) => dispatch => {
```
you can add REGISTER_PENDING, REGISTER_SUCESS and REGISTER_ERROR actions in order to control redux from redux dev tools and add some loading icon

InnaKutsik commented on src/actions/authentication.js:
```diff
@@ -0,0 +1,48 @@
+            });
+}
+
+export const loginUser = (user) => dispatch => {
```
you can add LOGIN_PENDING, LOGIN_SUCESS and LOGIN_ERROR actions in order to control redux from redux dev tools and add some loading icon

InnaKutsik commented on src/actions/authentication.js:
```diff
@@ -0,0 +1,48 @@
+    axios.post('/api/users/register', user)
+            .then(res => history.push('/login'))
+            .catch(err => {
+                dispatch({
```
move to action creater

roman454647 replied:
moved

InnaKutsik commented on src/actions/authentication.js:
```diff
@@ -0,0 +1,48 @@
+                dispatch(setCurrentUser(decoded));
+            })
+            .catch(err => {
+                dispatch({
```
move to action creater

roman454647 replied:
moved

InnaKutsik commented on src/Navbar/index.jsx:
```diff
@@ -0,0 +1,66 @@
+     </div>
+      <nav>
+      <div id="navig" className="navigation">
+        <a href="#Home">Home</a>
```
you have a navbar json

why dont use it here in map?

roman454647 replied:
used

roman454647 committed:
fixed  comments

i haven`t changed the eslintrc becouse  in project i have  a server side which still have a lot error so it would not pass landing,i will change it as soon as i change server side code
Also i need to rewrite menu o array

roman454647 committed:
update of missing var ,media query ,etc . Also some answer for some Inna coments.

roman454647 committed:
new update ,components was rewrited to array

to do
- eslinter
-and move some function to -actionCreator
-small refactoring of code(cntrl-alt-f

roman454647 committed:
new update

done
-menu and navbar was rewrited to array
- some changes in backend folder
-all import  images were moved to one folder

roman454647 committed:
some new update

anna-kuzii commented on backend/app.js:
```diff
@@ -0,0 +1,31 @@
+app.use('/api/users', users);
+
+app.get('/', (req, res) => {
+  res.send('Hello');
```
Why "hello"?

roman454647 replied:
just usual start of application

anna-kuzii commented on src/About/style.scss:
```diff
@@ -0,0 +1,98 @@
+    text-align: center;
+
+    @media screen and (max-width: $max-md-screen) {
+      p {
```
p inside p?

roman454647 replied:
not now

anna-kuzii commented on src/About/style.scss:
```diff
@@ -0,0 +1,98 @@
+    }
+  }
+
+  @media screen and (max-width: $max-md-screen) {
```
should be exactly next inside selector. it is hard to find what selector for?

anna-kuzii commented on src/About/style.scss:
```diff
@@ -0,0 +1,98 @@
+}
+
+
+.About {
```
About  or about?

roman454647 replied:
i have removed it

anna-kuzii commented on src/About/style.scss:
```diff
@@ -0,0 +1,98 @@
+}
+
+
+
```
so much spaces

roman454647 replied:
deleted

anna-kuzii commented on src/BookTable/index.jsx:
```diff
@@ -0,0 +1,74 @@
+}
+  from '../variables/export-img';
+
+export default () => (
```
no name?

roman454647 replied:
now it has name

anna-kuzii commented on src/BookTable/style.scss:
```diff
@@ -0,0 +1,124 @@
+  @media screen and (max-width: $max-xs-screen) {
+    display: none;
+  }
+  display: flex;
```
main styles should be before media queries

roman454647 replied:
done

anna-kuzii commented on src/Footer/index.jsx:
```diff
@@ -0,0 +1,52 @@
+}
+  from '../variables/export-img';
+
+const Footer = () => (
```
export here. use one convention

roman454647 replied:
i have used one convention,when we export afrer component

anna-kuzii commented on src/Footer/style.scss:
```diff
@@ -0,0 +1,72 @@
+  }
+}
+
+.About-footer {
```
About or about?
use one convention

roman454647 replied:
changed

anna-kuzii commented on src/Footer/style.scss:
```diff
@@ -0,0 +1,72 @@
+  margin-left: 100px;
+}
+
+.Our-Location {
```
Our or our
use one convention

roman454647 replied:
changed

anna-kuzii commented on src/Footer/style.scss:
```diff
@@ -0,0 +1,72 @@
+}
+
+
+div {
```
for all divs in the project?

roman454647 replied:
i cheked that item so i  have removed it

anna-kuzii commented on src/Footer/style.scss:
```diff
@@ -0,0 +1,72 @@
+    }
+  }
+
+.About {
```
About or about?

roman454647 replied:
changed

anna-kuzii commented on src/Ingredients/style.scss:
```diff
@@ -0,0 +1,80 @@
+  padding-bottom: 100px;
+  background: url('../image/bg4.png') no-repeat 73%;
+
+  @media screen and (max-width: $l-screen) {
```
correct var?

roman454647 replied:
changed

anna-kuzii commented on src/Login/index.jsx:
```diff
@@ -0,0 +1,117 @@
+      email: '',
+      password: '',
+      errors: {}
+        }
```
pls use hotkey to make code readable

anna-kuzii commented on src/Login/index.jsx:
```diff
@@ -0,0 +1,117 @@
+      errors: {}
+        }
+        this.handleInputChange = this.handleInputChange.bind(this);
+        this.handleSubmit = this.handleSubmit.bind(this);
```
in one component u use bind in constructor in another in render
pls use one convention

roman454647 replied:
ok

anna-kuzii commented on src/Login/style.scss:
```diff
@@ -0,0 +1,8 @@
+.container {
+  width: 700px;
```
hoe about mobile? will I see horizontal scroll?

roman454647 replied:
i cahange it to more suitable

anna-kuzii commented on src/Menu/AppetisersList.js:
```diff
@@ -0,0 +1,23 @@
+export default [
```
json?

roman454647 replied:
array

anna-kuzii commented on src/Menu/index.jsx:
```diff
@@ -0,0 +1,47 @@
+import MainList from './MainList';
+import PartOfMenu from '../PartOfMenu';
+
+const Menu = () => (
```
export

roman454647 replied:
should i export all function in same line woth the funcion initialising

anna-kuzii commented on src/Menu/style.scss:
```diff
@@ -0,0 +1,105 @@
+  margin-right: 0;
+}
+
+.Salads {
```
S or s

roman454647 replied:
small

anna-kuzii commented on src/Menu/style.scss:
```diff
@@ -0,0 +1,105 @@
+  padding-top: 75px;
+}
+
+.Dishes {
```
D or d

roman454647 replied:
change

anna-kuzii commented on src/Register/style.scss:
```diff
@@ -0,0 +1,8 @@
+.container {
+  width: 700px;
```
I ve already seen it

roman454647 replied:
changed

anna-kuzii commented on src/Rewiews/index.jsx:
```diff
@@ -0,0 +1,28 @@
+import './style.scss';
+import { divSlider } from '../variables/export-img';
+
+const Rewiews = () => (
```
export

roman454647 replied:
eslint want to export like a default  , but arrow  function need to be unnamed for default export ,so my decision was to  use export default NameFunction

anna-kuzii commented on src/Rewiews/style.scss:
```diff
@@ -0,0 +1,67 @@
+  }
+}
+
+.Rewiews {
```
R or r?

roman454647 replied:
r

roman454647 committed:
update some issues
